Feature/can i play with madness

geekbeast opened the pull request.

geekbeast committed:
Refactor organization class

- Convert to to Kotlin
- Add in new fields for auto-enroll and auto-grant

geekbeast committed:
Fix compile

geekbeast committed:
Fix compile

geekbeast committed:
Cleanup

geekbeast committed:
wowsers

geekbeast committed:
Compile

geekbeast committed:
Clean up

geekbeast committed:
Merge branch 'develop' into feature/can-i-play-with-madness

* develop:
  bugfix in path and return object

geekbeast committed:
Implement grants.

geekbeast committed:
Implement generation

geekbeast committed:
compile

katherinebernstein commented on src/main/kotlin/com/openlattice/organizations/Grant.kt:
```diff
@@ -0,0 +1,23 @@
+    Group,
+    Claim,
+    /**
+     *
```
is there supposed to be a comment here?

katherinebernstein commented on src/main/kotlin/com/openlattice/organizations/Organization.kt:
```diff
@@ -0,0 +1,145 @@
+        val smsEntitySetInfo: MutableSet<SmsEntitySetInformation>,
+        @JsonProperty(SerializationConstants.PARTITIONS) val partitions: MutableList<Int> = mutableListOf(),
+        @JsonProperty(SerializationConstants.APPS) val apps: MutableSet<UUID> = mutableSetOf(),
+        val appConfigs: Map<UUID, Map<UUID, AppTypeSetting>> = mutableMapOf(),
```
What is this field for? /does it need a @JsonProperty annotation?

geekbeast replied:
It was there before and wasn't used. I think I kept it there for you to use for your new app stuff, but I can remove if that is completely separate from organizations.

katherinebernstein replied:
I think think it's there on current develop 🤔

katherinebernstein replied:
At least it's not in Organization.java

geekbeast committed:
Bump spotbugs version.

geekbeast committed:
Resolve merge conflict.

geekbeast committed:
cleanup

geekbeast committed:
M-erge branch 'feature/can-i-play-with-madness' of github.com:openlattice/api into feature/can-i-play-with-madness

* 'feature/can-i-play-with-madness' of github.com:openlattice/api:
  Bump spotbugs version.
  incomplete metadata
  add test entitytypepropertymetadata
  bugfix type not set

UnsungHero97 commented on src/main/java/com/openlattice/organization/OrganizationsApi.java:
```diff
@@ -240,15 +246,18 @@ Void updateRoleDescription(
             @Body String description );
 
+    @PUT( BASE + ID_PATH + PRINCIPALS + ROLES + ROLE_ID_PATH + GRANT )
+    Void updateRoleAutoGrant(
```
Shouldn't this be `updateRoleGrant`? Why `AutoGrant`?

geekbeast committed:
Generate license reports.

geekbeast committed:
Remove unneeded field.

geekbeast committed:
Fix name.

geekbeast committed:
rename.

geekbeast committed:
Add connections.

UnsungHero97 commented on src/main/java/com/openlattice/organization/OrganizationsApi.java:
```diff
@@ -258,4 +260,13 @@ Void removeRoleFromUser(
             @Path( ROLE_ID ) UUID roleId,
             @Path( USER_ID ) String userId );
 
+    //Admin-only APIs
```
Any reason not to put these in `AdminApi`?

UnsungHero97 commented on src/main/java/com/openlattice/organization/OrganizationsApi.java:
```diff
@@ -258,4 +260,13 @@ Void removeRoleFromUser(
 
+    //Admin-only APIs
+    @POST( BASE + ID_PATH + CONNECTIONS )
+    Void addConnection( @Path( ID ) UUID organizationId, @Body Set<String> connections );
```
Will this endpoint add multiple connections if `Set<String> connections` has multiple values? If yes, can this be renamed to `addConnections`?

UnsungHero97 commented on src/main/kotlin/com/openlattice/organizations/Organization.kt:
```diff
@@ -0,0 +1,144 @@
+            @JsonProperty(SerializationConstants.SMS_ENTITY_SET_INFO)
+            smsEntitySetInfo: Optional<MutableSet<SmsEntitySetInformation>>,
+            @JsonProperty(SerializationConstants.PARTITIONS) partitions: Optional<MutableList<Int>>,
+            @JsonProperty(SerializationConstants.CONNECTIONS) enrollments: MutableSet<String> = mutableSetOf(),
```
`enrollments` > `connections`

geekbeast committed:
Cleanup.

geekbeast committed:
Documentation.

geekbeast committed:
Make it compile!

geekbeast committed:
TDF

geekbeast committed:
Resolve pr feedback

geekbeast committed:
move default parameter to end.

geekbeast committed:
Set default value

geekbeast committed:
Fix indent

geekbeast committed:
Add auth0 deps for now.

UnsungHero97 commented on src/main/kotlin/com/openlattice/organizations/Grant.kt:
```diff
@@ -0,0 +1,66 @@
+ */
+data class Grant(
+        @JsonProperty(SerializationConstants.GRANT_TYPE) val grantType: GrantType,
+        @JsonProperty(SerializationConstants.MAPPINGS) val settings: Set<String>,
```
`SerializationConstants.MAPPINGS` > `SerializationConstants.SETTINGS`

geekbeast committed:
Resolve and improve

geekbeast committed:
Moar grants.

geekbeast committed:
fix tdf

Commit:
OrganizationPrincipal implements Serializable

geekbeast committed:
Moar e-mail validation

geekbeast committed:
Move utils.

geekbeast committed:
Merge branch 'feature/can-i-play-with-madness' of github.com:openlattice/api into feature/can-i-play-with-madness

* 'feature/can-i-play-with-madness' of github.com:openlattice/api:
  OrganizationPrincipal implements Serializable

geekbeast committed:
fix compile

geekbeast committed:
Shouldn't be serializable

geekbeast committed:
Fix compile
